app/src/main/java/com/example/tasky/agenda/presentation/viewmodel/AgendaViewModel.kt

```diff
@@ -26,33 +28,120 @@ class AgendaViewModel @Inject constructor(
     private val agendaRepository: AgendaRepository,
 ) : ViewModel() {
 
+    private val selectedDate: MutableStateFlow<LocalDate> = MutableStateFlow(LocalDate.now())
+
     var state by mutableStateOf(AgendaState())
         private set
 
     init {
         state = state.copy(
-            days = dateGenerator.getWeek(),
             userInitials = GetInitialsUseCase(userCache.getUser()?.fullName),
-            selectedMonth = dateGenerator.getMonth()
         )
-        val now = LocalDate.now()
         viewModelScope.launch {
-            agendaRepository.getAgenda(now).collectLatest {
+            selectedDate.collect {
+                getAgendaForDate(it, shouldFetch = true)
                 state = state.copy(
-                    agendaItems = AddNeedleToAgenda(now.toEpochDay(), it.toMutableList())
+                    days = dateGenerator.getWeek(it),
+                    selectedMonth = dateGenerator.getMonth(it)
                 )
             }
         }
+    }
+
+    private fun getAgendaForDate(date: LocalDate, shouldFetch: Boolean = false) {
         viewModelScope.launch {
-            val fetchSuccessfully = agendaRepository.fetchAgenda(now)
+            agendaRepository.getAgenda(date).collect {
+                state = state.copy(
+                    agendaItems = AddNeedleToAgenda(date.toEpochDay(), it.toMutableList())
```

Comment:
`AddNeedleToAgenda` sounds like a function, but it's actually a class 🤔 
What speaks against just making this a normal function here?